### PR TITLE
COM-85: Refactor theming of ErrorBoundary

### PR DESCRIPTION
**Pull request by jamesricky**
Refactoring to a function-component is not possible, as there is no alternative to `componentDidCatch`, 
according to the [React Docs](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary):

> There is currently no way to write an error boundary as a function component.

### CODE REVIEW

File: packages/admin/admin/src/error/errorboundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -17,118 +17,213 @@ export type ErrorBoundaryClassKey =
...
 
                     {process.env.NODE_ENV === "development" && (
-                        <details className={classes.exceptionDetails}>
-                            <summary className={classes.exceptionSummary}>
-                                <div className={`${classes.exceptionSummaryIcon} ${classes.exceptionSummaryIconOpened}`}>
+                        <ExceptionDetails
+                            onToggle={(e) => {
+                                const showDetails = e.currentTarget.hasAttribute("open");
+                                this.setState((prev) => ({ ...prev, showDetails }));
+                            }}
```
Comment (johnnyomair) [inline packages/admin/admin/src/error/errorboundary/ErrorBoundary.tsx:203]: This is a new functionality, right? If yes, why did you add it? Could you please provide a screencast?

Comment (jamesricky) [reply packages/admin/admin/src/error/errorboundary/ErrorBoundary.tsx:203]: This doesn't change the behavior. 

Previously the icons where shown/hidden in the styling by selecting `details[open] .{icon-class}`. 
Like this:

```ts
exceptionDetails: {
    "&[open]": {
        "& $exceptionSummaryIconClosed": {
            display: "flex",
        },
        "& $exceptionSummaryIconOpened": {
            display: "none",
        },
    },
},
```

This is no longer possible, because the Details and the Icon are separate slots which cannot reference another slot, like you could do in styled-components for example:

```ts
const IconSlot = styled(...);
const DetailsSlot = styled(...)(css`
  &[open] ${IconSlot} { // Selecting `IconSlot` is not supported by MUIs `styled()`
    // show or hide
  }
`)
```

That's why I added an additional `open` value to the state, that just mirrors the `open` attribute on the `Details` element which changes on `onToggle`.

**Review by johnnyomair — APPROVED** [review]
